src/main/java/org/jfrog/hudson/ArtifactoryBuilder.java

```diff
@@ -381,16 +382,32 @@ private boolean isEmptyUrls(List<JFrogPlatformInstance> jfrogInstances) {
             return false;
         }
 
-        private boolean fillEmptyServers(List<JFrogPlatformInstance> jfrogInstances) {
+        private void autoFillPlatformServers(List<JFrogPlatformInstance> jfrogInstances) {
             if (jfrogInstances == null) {
-                return false;
+                return;
             }
             for (JFrogPlatformInstance instance : jfrogInstances) {
                 if (StringUtils.isBlank(instance.getArtifactoryServer().getArtifactoryUrl())) {
                     instance.getArtifactoryServer().setArtifactoryUrl(instance.getUrl() + "/artifactory");
+                    continue;
+                }
+                if (StringUtils.isBlank(instance.getUrl())) {
```

Comment:
Shouldn't this check be the first inside the for loop?
If ```instance.getUrl()``` is blank, then the Artifactory URL set in line 391 will be invalid.